python/cudf/cudf/pylibcudf_tests/test_json.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+import io
+
+import pandas as pd
+import pyarrow as pa
+import pytest
+from utils import sink_to_str
+
+import cudf._lib.pylibcudf as plc
+
+
+@pytest.mark.parametrize("rows_per_chunk", [8, 100])
+@pytest.mark.parametrize("lines", [True, False])
+def test_write_json_basic(table_data, source_or_sink, lines, rows_per_chunk):
+    plc_table_w_meta, pa_table = table_data
+    sink = source_or_sink
+
+    kwargs = dict()
+    if rows_per_chunk <= plc_table_w_meta.tbl.num_rows():
+        kwargs["rows_per_chunk"] = rows_per_chunk
+
+    plc.io.json.write_json(
+        plc.io.SinkInfo([sink]), plc_table_w_meta, lines=lines, **kwargs
+    )
+
+    # orient=records (basically what the cudf json writer does,
+    # doesn't preserve colnames when there are zero rows in table)
+    exp = pa_table.to_pandas()
+
+    if len(exp) == 0:
+        exp = pd.DataFrame()
```

Comment:
*question*: Is this a mismatch between cudf and pandas? If yes, we need to report a bug and fix it on the cudf side.

Reply:
No, orient="records" is just lossy.

e.g. 

```
>>> pd.DataFrame(columns=["a", "b"]).to_json(orient="records")
'[]'
```

Reply:
Actually, this is outdated from when I was trying to round-trip.

Let me try removing.